### PR TITLE
Role based spring security

### PR DESCRIPTION
**Pull request by RussellSk**

### CODE REVIEW

File: sandbox-backend/src/main/resources/db/changelog/changes/20210427_delete_data_from_permission.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+databaseChangeLog:
+  - changeSet:
+      id: 20210427_truncate_data_from_permission
+      author: skaldin
+      preCondition:
+        onFail: CONTINUE
```
Comment (ikantsevich) [inline sandbox-backend/src/main/resources/db/changelog/changes/20210427_delete_data_from_permission.yaml:6]: if we use CONTINUE then when the precondition was failed, the changeset will be run anyway and as a result, we get an error: table does not exist. It's the same in other places

File: sandbox-backend/src/main/resources/db/changelog/changes/20210425_add_column_to_permission.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            tableName: PERMISSION
+            columnName: PMN_ROUTE
+      changes:
+        - addColumn:
+            tableName: PERMISSION
+            columns:
+              name: PMN_ROUTE
+              type: VARCHAR(255)
+              afterColumn: PMN_NAME
+              constraints:
```
Comment (ikantsevich) [inline sandbox-backend/src/main/resources/db/changelog/changes/20210425_add_column_to_permission.yaml:18]: If we add a new column with notnull we should handle existing records, maybe add default value or set some value for old records

File: sandbox-backend/src/main/resources/db/changelog/changes/20210425_add_column_method_to_permission.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            tableName: PERMISSION
+            columnName: PMN_HTTP_METHOD
+      changes:
+        - addColumn:
+            tableName: PERMISSION
+            columns:
+              name: PMN_HTTP_METHOD
+              type: VARCHAR(255)
+              afterColumn: PMN_ROUTE
+              constraints:
```
Comment (ikantsevich) [inline sandbox-backend/src/main/resources/db/changelog/changes/20210425_add_column_method_to_permission.yaml:18]: the same

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/security/SecurityConstants.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.exadel.team2.sandbox.configuration.security;
+
+public final class SecurityConstants {
+    public static final String SECRET = "iazMqvVFVf9JaI8SHUA8O7uM1ZnEztXktOHUujxzrmTCT46Q6iCIk1l8A4NZ9Qw";
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/security/SecurityConstants.java:4]: Do we need to hardcode this value?

Comment (RussellSk) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/security/SecurityConstants.java:4]: No, can I move it to application.properties ?

File: sandbox-backend/src/main/resources/db/changelog/changes/20210427_delete_data_from_permission.yaml (hunk 1 above)
Comment (RussellSk) [reply sandbox-backend/src/main/resources/db/changelog/changes/20210427_delete_data_from_permission.yaml:6]: I got it, thanks

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/security/SecurityConstants.java (hunk 4 above)
Comment (RussellSk) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/security/SecurityConstants.java:4]: Done, move params to application.settings.

File: sandbox-backend/src/main/resources/db/changelog/changes/20210427_delete_data_from_permission.yaml (hunk 1 above)
Comment (RussellSk) [reply sandbox-backend/src/main/resources/db/changelog/changes/20210427_delete_data_from_permission.yaml:6]: I put, "onFail: HALT" here

**Review by ikantsevich — APPROVED** [review]

**Pull request merged**